### PR TITLE
#159206059 User can get notifications

### PR DESCRIPTION
**Pull request by anneKay**
#### What does this PR do?
- Send E-mail notifications to users
- Allow in-app notification

#### Description of Task to be completed?
- Get users following an author
- Create an email template for users
- Send email notification to users using `nodemailer`
- Set up firebase connection
- Set up notification database  on firebase
- Populate firebase database with notifications

#### How should this be manually tested?
In postman navigate to;
 (POST) `http://api/profiles/:username/follow` - follow a user
(POST) `http://api/articles/:slug/like` -like an article
(POST) `http://localhost:3000/api//articles/:slug/comments` - post a comment 
(POST) `http://localhost:3000/api/articles` - create article 

#### Any background context you want to provide?
- Pull in changes and run `npm install`
- Start server with `npm run dev`

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/159206059

### CODE REVIEW

**Review by Oloyedesinmiloluwa — APPROVED** [review]

**Review by veeqtor — APPROVED** [review]

File: controllers/UsersController.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import bcrypt from 'bcrypt';
 import jwt from 'jsonwebtoken';
 import { Op } from 'sequelize';
+// import db from '../models';
```
Comment (anuonifade) [inline controllers/UsersController.js]: Please remove commented code

File: config/serviceAccount.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export default {
+  type: 'service_account',
+  project_id: process.env.PROJECT_ID,
+  private_key_id: process.env.PRIVATE_KEY_ID,
+  private_key: process.env.PRIVATE_KEY.replace(/\\n/g, '\n'),
```
Comment (anuonifade) [inline config/serviceAccount.js:5]: Any reason for this?

Comment (anneKay) [reply config/serviceAccount.js:5]: Yes, the firebase private_key has many line breaks which Travis ignores and adds another backward slash, so this throws an error on Travis during the build but when you call the replace method on the private_key with a line break, it now applies the line breaks.

**Review by anuonifade — APPROVED** [review]
LGTM

**Pull request merged**